src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2337,6 +2337,36 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             }
             break;
         }
+        case SMART_ACTION_RANDOM_SOUND:
+        {
+            std::vector<uint32> sounds;
+
+            for (uint8 i = 0; i < SMART_ACTION_PARAM_COUNT - 1; i++)
+            {
+                if (e.action.randomSound.sound[i])
+                    sounds.push_back(e.action.randomSound.sound[i]);
+            }
+
+            bool onlySelf = e.action.randomSound.onlySelf != 0;
+
+            ObjectList* targets = GetTargets(e, unit);
+            if (targets)
+            {
+                for (ObjectList::const_iterator itr = targets->begin(); itr != targets->end(); ++itr)
+                {
+                    if (IsUnit(*itr))
+                    {
+                        uint32 sound = sounds[urand(0, sounds.size() - 1)];
+                        (*itr)->PlayDirectSound(sound, e.action.randomSound.onlySelf ? (*itr)->ToPlayer() : nullptr);
```

Comment:
Yes, replace `sound, e.action.randomSound.onlySelf` with `onlySelf` so you don't get the error `unused variable 'onlySelf'`, whcih Travis complains about. :)